index-import.cpp

```diff
@@ -242,7 +225,30 @@ static bool cloneRecords(StringRef recordsDirectory,
 int main(int argc, char **argv) {
   cl::ParseCommandLineOptions(argc, argv);
 
-  auto remapper = Remapper::createFromCommandLine(PathRemaps);
+  Remapper remapper;
+  // Parse the the path remapping command line flags. This converts strings of
+  // "X=Y" into a (regex, string) pair. Another way of looking at it: each
+  // remap is equivalent to the s/pattern/replacement/ operator.
+  auto errors = 0;
+  for (const auto &remap : PathRemaps) {
+    auto divider = remap.find('=');
+    auto pattern = remap.substr(0, divider);
+    std::regex re;
+    try {
+      re = std::regex(pattern);
+    } catch (const std::regex_error &e) {
+      errs() << "Error parsing regular expression: '" << pattern << "':\n"
+             << e.what() << "\n";
+      errors++;
+    }
+    auto replacement = remap.substr(divider + 1);
+    remapper.addRemap(re, replacement);
```

Comment:
Move these two lines into the `try` block? The `catch` doesn't `continue`, and so these lines are hit even when there's an error. The `try` block could be self contained:

```cpp
try {
  std::regex re{pattern};
  auto replacement = remap.substr(divider + 1);
  remapper.addRemap(re, replacement);
} catch...
```

Reply:
PS. Nice subtle improvement of `pattern`/`re` naming.

Reply:
Done